inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java

```diff
@@ -198,50 +222,201 @@ public void add(int auditID, String auditTag, String inlongGroupID, String inlon
         keyJoiner.add(String.valueOf(auditID));
         keyJoiner.add(auditTag);
         keyJoiner.add(String.valueOf(auditVersion));
-        addByKey(keyJoiner.toString(), count, size, delayTime);
+        addByKey(DEFAULT_ISOLATE_KEY, keyJoiner.toString(), count, size, delayTime);
+    }
+
+    public void add(AuditDimensions dimensions, AuditValues values) {
+        StringJoiner keyJoiner = new StringJoiner(FIELD_SEPARATORS);
+        keyJoiner.add(String.valueOf(dimensions.getLogTime() / PERIOD));
+        keyJoiner.add(dimensions.getInlongGroupID());
+        keyJoiner.add(dimensions.getInlongStreamID());
+        keyJoiner.add(String.valueOf(dimensions.getAuditID()));
+        keyJoiner.add(dimensions.getAuditTag());
+        keyJoiner.add(String.valueOf(dimensions.getAuditVersion()));
+        addByKey(dimensions.getIsolateKey(), keyJoiner.toString(), values.getCount(),
+                values.getSize(), values.getDelayTime());
     }
 
     /**
      * Add audit info by key.
      */
-    private void addByKey(String key, long count, long size, long delayTime) {
-        if (countMap.get(key) == null) {
-            countMap.put(key, new StatInfo(0L, 0L, 0L));
+    private void addByKey(long isolateKey, String statKey, long count, long size, long delayTime) {
+        if (null == this.preStatMap.get(isolateKey)) {
+            this.preStatMap.putIfAbsent(isolateKey, new ConcurrentHashMap<>());
         }
-        countMap.get(key).count.addAndGet(count);
-        countMap.get(key).size.addAndGet(size);
-        countMap.get(key).delay.addAndGet(delayTime);
+        ConcurrentHashMap<String, StatInfo> statMap = this.preStatMap.get(isolateKey);
+        if (null == statMap.get(statKey)) {
+            statMap.putIfAbsent(statKey, new StatInfo(0L, 0L, 0L));
+        }
+        StatInfo stat = statMap.get(statKey);
+        stat.count.addAndGet(count);
+        stat.size.addAndGet(size);
+        stat.delay.addAndGet(delayTime);
     }
 
     /**
-     * Send audit data
+     * Flush audit data by default audit version
      */
-    public synchronized void send() {
+    public synchronized void flush() {
+        flush(DEFAULT_AUDIT_VERSION);
+    }
+
+    /**
+     * Flush audit data
+     */
+    public synchronized void flush(long isolateKey) {
+        if (flushTime.putIfAbsent(isolateKey, Calendar.getInstance().getTimeInMillis()) != null
```

Comment:
"Calendar.getInstance().getTimeInMillis()" is same with "System.currentTimeMillis". They both get TimeZone +00:00.